modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -15,52 +15,91 @@ import Tools from '../api/util/Tools';
 import { isWsPreserveElement } from '../dom/ElementType';
 import * as TrimHtml from '../dom/TrimHtml';
 import * as Zwsp from '../text/Zwsp';
-import { Content, ContentFormat, GetContentArgs } from './ContentTypes';
+import { Content, GetContentArgs } from './ContentTypes';
+import { GetContentFormatter } from './EditorContent';
+
+const defaultContentFormatter = (editor: Editor, args: GetContentArgs): Content =>
+  Optional.from(editor.getBody())
+    .fold(
+      Fun.constant(args.format === 'tree' ? new AstNode('body', 11) : ''),
+      (body) => getContentFromBody(editor, args, body)
+    );
+
+const addDefaultGetFormats = (formatCell: Cell<Record<string, GetContentFormatter>>) => {
+  addGetContentFormatter('raw', defaultContentFormatter, formatCell);
+  addGetContentFormatter('text', defaultContentFormatter, formatCell);
+  addGetContentFormatter('html', defaultContentFormatter, formatCell);
+  addGetContentFormatter('tree', defaultContentFormatter, formatCell);
```

Comment:
This is unnecessarily duplicated... we also don't need to use the same formatter for `tree` now so can we please make it a separate formatter instead?

For the duplication, this could be changed to something like this:

```
Arr.each([ 'raw', 'text', 'html' ], (format) {
  addGetContentFormatter(format, defaultContentFormatter, formatCell);
});
addGetContentFormatter('tree'', treeContentFormatter, formatCell);
```

Reply:
Made the changes, though I think they feel a lot less safe than the old version, at least in the setContentImpl.

Reply:
Yeah, you've broken a few things here now, as the relevant events aren't firing for the tree implementation. You've also not split it properly either, as take `defaultContentFormatter` for example it still has a `args.format === 'tree'` check... so it probably needs to be something like this instead:

```
const defaultContentFormatter = (editor: Editor, args: GetContentArgs): string =>
  Optional.from(editor.getBody()).fold(
    Fun.constant(''),
    (body) => getContentFromBody(editor, args, body)
  );

const treeContentFormatter = (editor: Editor, args: GetContentArgs): AstNode =>
  Optional.from(editor.getBody()).fold(
    Fun.constant(new AstNode('body', 11)),
    (body) => getContentFromBody(editor, args, body)
  );
```

- [ ] Fix events not firing for the tree implementation